esphome/components/mdns/__init__.py

```diff
@@ -101,13 +101,20 @@ async def to_code(config):
     await cg.register_component(var, config)
 
     for service in config[CONF_SERVICES]:
-        txt = [
-            mdns_txt_record(txt_key, txt_value)
-            for txt_key, txt_value in service[CONF_TXT].items()
-        ]
-
+        txt = []
+        args = []
+        for txt_key, txt_value in service[CONF_TXT].items():
+            template_txt_value_ = await cg.templatable(txt_value, args, cg.std_string)
+            txt.append(
+                cg.StructInitializer(
+                    MDNSTXTRecord,
+                    ("key", txt_key),
+                    ("value", template_txt_value_),
+                )
+            )
+        template_port_value_ = await cg.templatable(service[CONF_PORT], args, cg.uint16)
```

Comment:
```suggestion
        txt = [
            cg.StructInitializer(
                MDNSTXTRecord,
                ("key", txt_key),
                ("value", await cg.templatable(txt_value, [], cg.std_string)),
            )
            for txt_key, txt_value in service[CONF_TXT].items()
        ]
        template_port_value_ = await cg.templatable(service[CONF_PORT], [], cg.uint16)
```

I was a bit confused as to why args was being reused between function because its never mutated by `cg.templatable` AFAICT